src/game/BattleGround/BattleGround.cpp

```diff
@@ -583,6 +583,9 @@ void BattleGround::CastSpellOnTeam(uint32 SpellID, Team teamId)
 
 void BattleGround::RewardHonorToTeam(uint32 Honor, Team teamId)
 {
+    if (teamId == TEAM_NONE)
+        return;
```

Comment:
Obviously the performance is negligible in all touched code but why intentionally write more inefficient and harder to understand code than my suggestion of moving it to the single needed place in EndBattleGround

Reply:
This is to make sure this call is future proof for other use cases and fits the design of the function which is `Reward Honor To Team` that implies a valid team which `TEAM_NONE` does not fit. This is not something that would be considered harder to understand.